### ISSUE TITLE
Potential issue on Windows?

### ISSUE THREAD

**Issue by simon-frankau** [MichaelMcDonnell/test_bin#6]
Hi.

I use `test_bin` in https://github.com/simon-frankau/galette (thanks, it's helpful!), and a user on Windows has found it failing to invoke a binary - https://github.com/simon-frankau/galette/issues/9 .

I would usually try to find a minimal reproduction before approaching dependency owners, but I don't have a Rust-on-Windows install, so before I invest a bunch of time trying to set up such a thing, I was wondering if `test_bin` was already known to work or not work on Windows?

Thanks!

**Comment by MichaelMcDonnell** [comment]
Sorry for the late response. I'm currently on hiatus from open source development. I sold my PCs and moved to Hawaii.  I won't be able to test on Windows until I get around buying a laptop again (I'm trying to hold out for the AMD Framework).  I can borrow my wife's old MacBook and merge any pull requests from that.

The code used to work on Windows but I might have broken it in a recent update. The user could try an older version and see if it works.

I don't have CI set up for testing on Windows. I also don't have CI set up for testing cross compiling (which was the most recent fix). Pull requests are welcome 😁

**Comment by MichaelMcDonnell** [comment]
Sorry, I didn't mean to close the issue (writing from my phone).

**Comment by simon-frankau** [comment]
Cheers. I do have a Windows machine (currently pretty much games-only), so I guess I'll have a go at installing the Rust toolchain on it and see if I can get you a pull request. :)

**Comment by MichaelMcDonnell** [comment]
Thanks! I've started working on Windows CI. I don't think it shouldn't be too difficult to add. I'm trying to re-use ripgrep's CI setup, but with fewer targets.

**Comment by MichaelMcDonnell** [comment]
I just added Windows to the CI. The Windows test [ran without an issue.](https://github.com/MichaelMcDonnell/test_bin/actions/runs/6230327325), so there's at least not anything super obvious wrong.

**Comment by MichaelMcDonnell** [comment]
@simon-frankau I read your integration test again and noticed that it tries to execute the `diff` command. It's not always installed on Windows, so I'm pretty sure that's the issue. I would recommend adding some more error handling in `check_output_matches`. Feel free to re-open if that's not the problem.

**Comment by simon-frankau** [comment]
Oh, I feel like such an idiot! Thank you, and sorry you ended up investigating my bug.

I'm used to debugging with a stack trace, but yes, I'll look into improving the error handling.

(Slow reply due to update getting caught in my spam filter.)

**Comment by MichaelMcDonnell** [comment]
No worries. I've made that kind of mistake so many times. It was confusing for me too because there were two different executables that were executed. Btw I think the GitHub Windows test runners have diff installed. I tried running your tests on them and they failed in a different place.

This issue helped move this project forward since I now have automated tests for Windows and cross. It was something I wanted to do, and you motivated me to do it, so thank you for that 😄
